common/api.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_COMMON_API_HPP
+#define MENDER_COMMON_API_HPP
+
+#include <string>
+#include <vector>
+
+#include <common/common.hpp>
+#include <common/expected.hpp>
+#include <common/json.hpp>
+
+namespace mender {
+namespace common {
+namespace api {
```

Comment:
Feel free to suggest a better name/place for this (and things like `OutgoingAPIRequest` in the future)

Reply:
First thing which struck me is: Should it not be under `http::`?

Reply:
You could argue for both -- does this code have to care it's HTTP? And also, will there be any other Mender API used here?

Reply:
Other than `HTTP` you mean?

Reply:
How about simply `mender::api`? Then we can put `mender::api::auth` under that. I think this is better to not overbloat the already very crowded `common` namespace. Plus, `api` is already pretty generic, but still a bit narrower than `common`.

I think I vote for not putting under `http`, because the transport protocol isn't really important here IMHO. And we could add other transport protocols with similar semantics for other platforms later.

Reply:
I think I `mender::api` is a good idea.  But it wouldn't hurt if it internally had `http`, and then the `mender::api::auth` could internally use `http::`

Reply:
I did this in the `auth` client now: https://github.com/mendersoftware/mender/pull/1278#pullrequestreview-1487616296

Reply:
The same in #1282. Adapting this PR now.